File: src/desktop-edge-agent/src/FccDesktopAgent.Core/Adapter/Common/IFccPeripheralMonitor.cs
namespace FccDesktopAgent.Core.Adapter.Common;

/// <summary>
/// Optional interface for adapters that receive peripheral device events
/// (EPT terminals, BNA devices, dispenser install data).
/// Ported from legacy DPP port 5006 peripheral message handling.
/// </summary>
public interface IFccPeripheralMonitor
{
    /// <summary>
    /// Request current peripheral device inventory from FCC.
    /// </summary>
    Task<PeripheralInventory> GetPeripheralInventoryAsync(CancellationToken ct);
}

/// <summary>Inventory of peripheral devices connected to the FCC.</summary>
public sealed record PeripheralInventory(
    IReadOnlyList<DispenserInfo> Dispensers,
    IReadOnlyList<EptTerminalInfo> EptTerminals);

/// <summary>Dispenser installation data. Ported from legacy DispenserInstallData.</summary>
public sealed record DispenserInfo(string DispenserId, string Model);

/// <summary>EPT (Electronic Payment Terminal) info. Ported from legacy EptInfo.</summary>
public sealed record EptTerminalInfo(string TerminalId, string Version);

/// <summary>BNA (Banknote Acceptor) report. Ported from legacy EptBnaReport.</summary>
public sealed record BnaReport(string TerminalId, int NotesAccepted, DateTimeOffset ReportedAtUtc);

File: src/edge-agent/app/src/main/kotlin/com/fccmiddleware/edge/adapter/common/IFccPeripheralMonitor.kt
package com.fccmiddleware.edge.adapter.common

/**
 * Optional interface for adapters that receive peripheral device events.
 * Ported from legacy DPP port 5006 peripheral message handling.
 */
interface IFccPeripheralMonitor {
    suspend fun getPeripheralInventory(): PeripheralInventory
}

data class PeripheralInventory(
    val dispensers: List<DispenserInfo>,
    val eptTerminals: List<EptTerminalInfo>,
)

data class DispenserInfo(val dispenserId: String, val model: String)
data class EptTerminalInfo(val terminalId: String, val version: String)
data class BnaReport(val terminalId: String, val notesAccepted: Int, val reportedAtUtc: String)
